kik_unofficial/datatypes/xmpp/chatting.py

```diff
@@ -3,113 +3,58 @@
 """
 
 import time
-import os
-import requests
-import json
-import base64
-from io import BytesIO
-from PIL import Image
+from typing import Union
+
 from bs4 import BeautifulSoup
+from lxml import etree
+from lxml.etree import Element
+
 from kik_unofficial.datatypes.peers import Group
-from kik_unofficial.datatypes.xmpp.base_elements import XMPPElement, XMPPResponse
-from kik_unofficial.utilities.parsing_utilities import ParsingUtilities
+from kik_unofficial.datatypes.xmpp.base_elements import XMPPElement, XMPPResponse, XMPPContentResponse, \
+    XMPPReceiptResponse, XMPPOutgoingContentMessageElement, XMPPOutgoingMessageElement, XMPPOutgoingIsTypingMessageElement
+from kik_unofficial.http_requests.tenor_client import KikTenorClient
+from kik_unofficial.utilities.parsing_utilities import ParsingUtilities, get_text_safe
 
 
-class OutgoingChatMessage(XMPPElement):
+class OutgoingChatMessage(XMPPOutgoingMessageElement):
     """
     Represents an outgoing text chat message to another kik entity (member or group)
     """
 
-    def __init__(self, peer_jid, body, is_group=False, bot_mention_jid=None):
-        super().__init__()
-        self.peer_jid = peer_jid
+    def __init__(self, peer_jid: str, body: str):
+        super().__init__(peer_jid)
         self.body = body
-        self.is_group = is_group
-        self.bot_mention_jid = bot_mention_jid
-
-    def serialize(self):
-        timestamp = str(int(round(time.time() * 1000)))
-        message_type = "groupchat" if self.is_group else "chat"
-        bot_mention_data = (
```

Comment:
Bot mention support has been removed as the official Kik Bot API has been discontinued, and sending this causes the message to be dropped by the Kik server.

Reply:
I understand. Where do you see that the official Kik bot API has been discontinued?

Reply:
Was first announced in https://kik.com/blog/kik-medialab-acquisition/

Then in early 2022 the API endpoints were shutdown, most bot accounts were banned, and it ceased to function.

Reply:
Even though [this page](https://www.kik.com/developers/) still states you can create bots,

Reply:
It's wrong / outdated. You can't create bots anymore

Reply:
I only see they wanted to "eradicate spam bots and unwanted messages"

Reply:
> For the time being, we will be discontinuing the video chat toggle as well as the 3rd party bots platform.

grep for it in https://kik.com/blog/kik-medialab-acquisition/

Reply:
OK, I see.